### PR TITLE
Removed Aspera, added boto3 + FIRE and some refactoring

### PR DESCRIPTION
**Pull request by mberacochea**

### CODE REVIEW

File: fetchtool/abstract_fetch.py [hunk 1]
```diff
@@ -148,9 +167,18 @@ def _parse_args(self, argv):
             help="Fixed runs in project description file",
             action="store_true",
         )
+        parser.add_argument("-e", "--ebi", required=False, help="Set this flag when running on EBI infrastcutrue", action="store_true")
```
Comment (SandyRogers) [inline fetchtool/abstract_fetch.py:170]:
```suggestion
        parser.add_argument("-e", "--ebi", required=False, help="Set this flag when running on EBI infrastructure", action="store_true")
```

**Review by SandyRogers — COMMENTED** [review]
Thanks @mberacochea !
I left a few small points/queries inline. Other than that, one day it would be nice to add a table of the different protocols and their pros/cons to the README.

File: fetchtool/abstract_fetch.py [hunk 2]
```diff
@@ -532,84 +562,45 @@ def download_lftp(self, dest, url):
...
+                    ftp.retrbinary("RETR " + file_name, output_file.write)
+                logging.info("File " + dest + " downloaded.")
+                return True
+        except ftplib.all_errors as e:
+            logging.error(e)
             return False
 
-    def download_aspera(self, dest: str, url: str) -> bool:
-        """Download using the aspera cli.
+    def download_aws(self, dest: str, url: str) -> bool:
```
[the remark this answers is not in the document]
Comment (mberacochea) [reply fetchtool/abstract_fetch.py:586]: it should be `download_fire`, not sure why I picked aws

File: README.md [hunk 3]
```diff
@@ -63,30 +56,17 @@ Install from the git repo
 $ pip install git+ssh://git@github.com/EBI-Metagenomics/fetch_tool.git
```
[the remark this answers is not in the document]
Comment (mberacochea) [reply README.md:56]: ah, cool. It probably is, and "safer"... not every env allows ssh connections to github or anywhere really

File: Containerfile [hunk 4]
```diff
@@ -1,21 +1,22 @@
-FROM python:3.9-slim
+FROM mambaorg/micromamba:1.5.8
```
[the remark this answers is not in the document]
Comment (mberacochea) [reply Containerfile:1]: It's very similar in size, 200MB~ish if I'm not mistaken. This is the base image seqera containers use, and biocontainers I think

**Pull request merged**